tests/unit/lms/product/product_test.py

```diff
@@ -0,0 +1,37 @@
+from dataclasses import dataclass, fields
+from unittest.mock import sentinel
+
+import pytest
+
+from lms.product.product import PluginConfig, Product
+
+
+class FakePlugin:
+    request: None
+
+    @classmethod
+    def from_request(cls, request):
+        instance = cls()
+        instance.request = request
+        return instance
```

Comment:
I absolutely could not get this to work with regular mocks of any kind. All sorts of crazy stuff happened. This works and I'm going to take a step back because I burned a lot of time on this, but if you can do better please do.

Reply:
This doesn't look horrible in any case. If you already have fought with it we can keep it until the next product refactor.